src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -396,53 +396,6 @@ def test_lfric_driver_operator():
     build.compile_file("driver-operator-test.F90")
 
 
-# ----------------------------------------------------------------------------
-@pytest.mark.parametrize("name, filename",
-                         [("x_innerproduct_x",
-                           "15.9.2_X_innerproduct_X_builtin.f90"),
-                          ("sum_x", "15.14.3_sum_setval_field_builtin.f90"),
-                          ("int_x", "15.10.3_int_X_builtin.f90"),
-                          ("real_x", "15.28.2_real_X_builtin.f90")
-                          ])
-def test_lfric_driver_unsupported_builtins(name, filename, capsys):
-    '''The following builtins do not have a proper lower_to_language_level
-    method to create the PSyIR, so they are not supported in the driver
-    creation: LFRicXInnerproductXKern, LFRicSumXKern, LFRicIntXKern,
-    LFRicRealXKern. This tests also the error handling of the functions
-    write_driver, get_driver_as_string, create. '''
-
-    _, invoke = get_invoke(filename, API, dist_mem=False, idx=0)
-
-    driver_creator = LFRicExtractDriverCreator()
-    read_write_info = ReadWriteInfo()
-
-    # The create method should raise an exception
-    # -------------------------------------------
-    with pytest.raises(NotImplementedError) as err:
-        # The parameters do not really matter
-        driver_creator.create(invoke.schedule, read_write_info, "extract",
-                              "_post", region_name=("region", "name"))
-    assert f"LFRic builtin '{name}' is not supported" in str(err.value)
-
-    # The get_driver_as_string method returns
-    # an empty string, but prints an error message
-    # --------------------------------------------
-    code = driver_creator.get_driver_as_string(invoke.schedule,
-                                               read_write_info, "extract",
-                                               "_post", ("region", "name"))
-    assert code == ""
-    out, _ = capsys.readouterr()
-    assert (f"Cannot create driver for 'region-name' because:\nLFRic builtin "
-            f"'{name}' is not supported" in out)
-
-    # write_driver prints an error message, and does not return an error
-    # ------------------------------------------------------------------
-    driver_creator.write_driver(invoke.schedule, read_write_info,
-                                "extract", "_post", ("region", "name"))
-    assert (f"Cannot create driver for 'region-name' because:\nLFRic builtin "
-            f"'{name}' is not supported" in out)
-
-
```

Comment:
This has left some lines uncovered, see https://app.codecov.io/gh/stfc/PSyclone/pull/2376/indirect-changes

But I think this checks can be removed now since from the lfric_extract_driver_creator.py since there are no more builtins without lowering?

Similarly the LFRicBulitIn.array_ref method is not used anymore, can we remove it?

Reply:
Oops, forgot these! Removed.

Reply:
There are still uncovered lines on `fric_extract_driver_creator.py`, but it seems we shouldn't delete any further. Specifically these:
```
 except NotImplementedError as err:
     print(f"Cannot create driver for '{region_name[0]}-"
           f"{region_name[1]}' because:")
     print(str(err))
     return ""
```

Maybe @hiker can advice if there are other ways to trigger this error message now that we don't have any Bultins that don't have a lowering method implemented?

Reply:
I believe the following code triggers that exception:
```
        for node in nodes:
            for builtin in node.walk(LFRicBuiltIn):
                # If the lower_to_language function is not overwritten from
                # the implementation in Node, the builtin is not yet supported:
                if type(builtin).lower_to_language_level == \
                        Node.lower_to_language_level:
                    raise NotImplementedError(
                        f"LFRic builtin '{builtin.name}' is not supported")
```
So I check if there is an overwritten `lower_to_language_level` function is available, and if not (i.e. only the base class) it raises an exception. With this PR, when all builtins are lowered, this test in the driver (and then the corresponding test in the tests directory) can be removed.

Reply:
What happens if this line is triggered, will psyclone completely abort? 

I added a print statement only so I could easily test as much of LFRic as possible, even if (say) a certain intrinsic is not supported. With some files containing 20+ invokes, if one kernel has a problem that the driver extraction cannot handle, we miss out a lot of additional tests.

I have opened #2120 to find a better way (command line option or so, so the user can control if it should abort or just continue with one missing driver - I think I need more command line options in general, so I'll do this as a separate commit), so from that point of view I would prefer the old behaviour (just print the message and return an empty string), and use #2120 to properly control what should happen in case of an error.

Reply:
I agree with @hiker on not changing the behavior on this PR.

The problem is that the error that was triggering this lines on the test (the builtin does not have a lowering method) does not exist anymore, so we need an alternative NotImplementedError for triggering them.

Alternatively, if it would be easier for hiker to work on a better solution, I would be ok on commenting out the uncovered lines (the print) and add a TODO to #2120

Reply:
Yes, I'm happy with that as well, thanks!

Reply:
OK, I restored the `NonImplementedError`, commented out `print` statements and added TODO as requested. Please note that I had to remove `as err` because `pylint` complained about `err` not being used after commenting out `print` statements. However, that does not solve the problem of coverage as the `except NonImplementedError` line is not being tested. I'm happy to restore `print` statements and live with a bit less coverage, TBH.